sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -368,6 +373,98 @@ public virtual async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream str
             }
         }
 
+        private async Task<ChunkedUploadResult> UploadInChunks(string location, Stream stream, int maxChunkSize, bool async = true, CancellationToken cancellationToken = default)
+        {
+            long length = stream.Length;
+
+            string digest = default;
+            ResponseWithHeaders<ContainerRegistryBlobUploadChunkHeaders> uploadChunkResult = null;
+
+            if (length < maxChunkSize)
+            {
+                digest = OciBlobDescriptor.ComputeDigest(stream);
+
+                uploadChunkResult = async ?
+                    await _blobRestClient.UploadChunkAsync(location, stream, cancellationToken: cancellationToken).ConfigureAwait(false) :
+                    _blobRestClient.UploadChunk(location, stream, cancellationToken: cancellationToken);
+
+                return new ChunkedUploadResult(digest,
+                    uploadChunkResult.Headers.Location,
+                    length);
+            }
+
+            byte[] buffer = new byte[maxChunkSize];
```

Comment:
Same issue as above, if user passes some arbitrary large number for `maxChunkSize`, then we will endup creating really large chunks and the user can run out of memory.

So I think we might consider setting up some upper limit for `maxChunkSize` to break the stream in chunks.

Reply:
Discussed here: https://github.com/Azure/azure-sdk-for-net/pull/32059#discussion_r1018332277